### PR TITLE
Modify rule S4125: Adapt to LaYC

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S4125/javascript/rule.adoc [hunk 1]
```diff
@@ -2,34 +2,36 @@
...
-* "boolean"
-* "number"
-* "string"
-* "symbol" (since ECMAScript 2015)
-* "function"
-* "object" (for ``++null++`` and any other object)
-* "BigInt" (since ECMAScript 2020)
-
-Compare a ``++typeof++`` expression to anything else, and the result is predefined: ``++false++``.
+* ++"undefined"++
```
Comment (victor-diez-sonarsource) [inline rules/S4125/javascript/rule.adoc:5]: what are the ++ doing? I see no impact of the result

File: rules/S4125/javascript/rule.adoc [hunk 2]
```diff
@@ -2,34 +2,36 @@
...
-function someFunc(x) {
+function isNumber(x) {
   return typeof x === "number";
 }
 ----
 
+== Resources
+=== Documentation
+
+* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof[MDN typeof]
```
Comment (victor-diez-sonarsource) [inline rules/S4125/javascript/rule.adoc:37]: this ref could be useful also: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#primitive_values

File: rules/S4125/javascript/rule.adoc [hunk 3]
```diff
@@ -2,34 +2,36 @@
...
+* ++"number"++
+* ++"string"++
+* ++"symbol"++ (since ECMAScript 2015)
+* ++"function"++
+* ++"object"++ (for ``++null++`` and any other object)
+* ++"bigint"++ (since ECMAScript 2020)
 
+Compare a ``++typeof++`` expression to anything else, and the result will always be ``++false++``.
 
 === Noncompliant code example
```
Comment (victor-diez-sonarsource) [inline rules/S4125/javascript/rule.adoc:16]: Remove noncompliant title

File: rules/S4125/javascript/rule.adoc [hunk 4]
```diff
@@ -2,34 +2,36 @@
...
 ----
-function someFunc(x) {
-  return typeof x === "Number"; // Noncompliant, function will always return 'false'
+function isNumber(x) {
+  return typeof x === "Number"; // Noncompliant: the function always returns 'false'
 }
 ----
 
-
 === Compliant solution
```
Comment (victor-diez-sonarsource) [inline rules/S4125/javascript/rule.adoc:25]: same here, remove compliant title and add some explanation about the correction: 

"Instead, make sure you are always comparing the against one of the seven possible values"

**Review by victor-diez-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S4125/javascript/rule.adoc (hunk 3 above)
Comment (yassin-kammoun-sonarsource) [reply rules/S4125/javascript/rule.adoc:16]: 👍

File: rules/S4125/javascript/rule.adoc (hunk 4 above)
Comment (yassin-kammoun-sonarsource) [reply rules/S4125/javascript/rule.adoc:25]: 👍

File: rules/S4125/javascript/rule.adoc (hunk 2 above)
Comment (yassin-kammoun-sonarsource) [reply rules/S4125/javascript/rule.adoc:37]: 👍

File: rules/S4125/javascript/rule.adoc (hunk 1 above)
Comment (yassin-kammoun-sonarsource) [reply rules/S4125/javascript/rule.adoc:5]: Forgot the backticks.

**Review by victor-diez-sonarsource — APPROVED** [review]
LGTM, but please change reference titles to always have a hyphen (`-`) between resource and title:

MDN - Primitive values

**Pull request merged**